src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Tests/Generated/Microsoft.Interop.ComInterfaceGenerator/Microsoft.Interop.ComInterfaceGenerator/ComInterfaceGenerator.Tests.IDerivedComInterface.cs

```diff
@@ -51,10 +81,84 @@
 
         return __retVal;
     }
+
+    [System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.Interop.ComInterfaceGenerator", "42.42.42.42")]
+    [System.Runtime.CompilerServices.SkipLocalsInitAttribute]
+    int global::ComInterfaceGenerator.Tests.IDerivedComInterface.GetData()
+    {
+        var(__this, __vtable_native) = ((System.Runtime.InteropServices.Marshalling.IUnmanagedVirtualMethodTableProvider)this).GetVirtualMethodTableInfoForKey(typeof(global::ComInterfaceGenerator.Tests.IDerivedComInterface));
+        int __retVal;
+        int __invokeRetVal;
+        {
+            __invokeRetVal = ((delegate* unmanaged<void*, int*, int> )__vtable_native[3])(__this, &__retVal);
+        }
+
+        // Unmarshal - Convert native data to managed data.
+        System.Runtime.InteropServices.Marshal.ThrowExceptionForHR(__invokeRetVal);
+        return __retVal;
+    }
+
+    [System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.Interop.ComInterfaceGenerator", "42.42.42.42")]
+    [System.Runtime.CompilerServices.SkipLocalsInitAttribute]
+    void global::ComInterfaceGenerator.Tests.IDerivedComInterface.SetData(int n)
+    {
+        var(__this, __vtable_native) = ((System.Runtime.InteropServices.Marshalling.IUnmanagedVirtualMethodTableProvider)this).GetVirtualMethodTableInfoForKey(typeof(global::ComInterfaceGenerator.Tests.IDerivedComInterface));
+        int __invokeRetVal;
+        {
+            __invokeRetVal = ((delegate* unmanaged<void*, int, int> )__vtable_native[4])(__this, n);
+        }
+
+        // Unmarshal - Convert native data to managed data.
+        System.Runtime.InteropServices.Marshal.ThrowExceptionForHR(__invokeRetVal);
+    }
 }
 
 file unsafe partial interface InterfaceImplementation
 {
+    [System.Runtime.InteropServices.UnmanagedCallersOnlyAttribute]
```

Comment:
We don't need to do anything about shadowing in the unmanaged->managed direction here, as we don't save much by avoiding a cast of a managed object to an interface type (unlike QueryInterface, this is cheap).

Reply:
I think I got rid of that in 3eba497.